ecs_pod_creator_test.go

```diff
@@ -705,6 +706,39 @@ func TestECSPodDefinition(t *testing.T) {
 
 			assert.NotEqual(t, h0, h1, "container repo creds name should affect hash")
 		})
+		t.Run("ChangesForDifferentLogConfiguration", func(t *testing.T) {
+			opts := getValidPodDefOpts()
+			creds := NewRepositoryCredentials().SetID("id")
+			opts.ContainerDefinitions[0].SetRepositoryCredentials(*creds)
+			assert.NotEqual(t, baseHash, opts.Hash(), "container repo creds should affect hash")
+		})
+		t.Run("ChangesForDifferentLogConfigurationDriver", func(t *testing.T) {
+			opts := getValidPodDefOpts()
+
+			logConf := NewLogConfiguration()
+			opts.ContainerDefinitions[0].SetLogConfiguration(*logConf)
+			h0 := opts.Hash()
+
+			opts.ContainerDefinitions[0].SetLogConfiguration(*logConf.SetLogDriver(awsECS.LogDriverAwslogs))
+			h1 := opts.Hash()
+
+			assert.NotEqual(t, h0, h1, "log configuration driver should affect hash")
+		})
+		t.Run("ChangesForDifferentLogConfigurationOptions", func(t *testing.T) {
+			opts := getValidPodDefOpts()
+
+			logConf := NewLogConfiguration()
+			opts.ContainerDefinitions[0].SetLogConfiguration(*logConf)
+			h0 := opts.Hash()
+
+			opts.ContainerDefinitions[0].SetLogConfiguration(*logConf.SetOptions(map[string]*string{"key": utility.ToStringPtr("value")}))
+			h1 := opts.Hash()
+
+			opts.ContainerDefinitions[0].SetLogConfiguration(*logConf.SetOptions(map[string]*string{"key": utility.ToStringPtr("value"), "key2": utility.ToStringPtr("value")}))
+			h2 := opts.Hash()
+
+			assert.NotEqual(t, h0, h1, h2, "log configuration options should affect hash")
+		})
```

Comment:
Can we add an equivalent test to `ReturnsSameValueForSameUnorderedTags` for `Options`?